api/src/main/java/org/msf/module/visit/CloseVisitOnAnOutcomeTask.java

```diff
@@ -49,8 +57,87 @@ public void execute() {
+        } else {
+            return (isNotInAnyConfiguredProgramState(programWorkflowService, activePatientProgramList.get(0)) || isBedAssigned(patient));
+        }
+    }
+
+    private boolean isNotInAnyConfiguredProgramStatesWithMultipleProgramsOrBedAssigned(ProgramWorkflowService programWorkflowService, List<PatientProgram> activePatientProgramList, Patient patient) {
+        int activeProgramsWithConfiguredState = 0;
+        for (PatientProgram patientProgram : activePatientProgramList) {
+            if (!isNotInAnyConfiguredProgramState(programWorkflowService, patientProgram)) {
+                activeProgramsWithConfiguredState++;
+            }
+        }
+        if (activeProgramsWithConfiguredState == activePatientProgramList.size()) {
+            return isBedAssigned(patient);
+        }
+        return true;
+    }
+
     private boolean outcomeAvailable(Visit openVisit, List<Concept> concepts) {
         BahmniObsService bahmniObsService = Context.getService(BahmniObsService.class);
         return !bahmniObsService.getLatestObsByVisit(openVisit, concepts, null, true).isEmpty();
     }
+
+    private List<PatientProgram> getActivePatientProgramForPatient(Patient patient, ProgramWorkflowService programWorkflowService) {
+        List<PatientProgram> patientPrograms = programWorkflowService.getPatientPrograms(patient, null, null, null, null, null, false);
+        List<PatientProgram> activePatientProgramList = new ArrayList<>();
+        for (PatientProgram patientProgram : patientPrograms) {
+            if (patientProgram.getDateCompleted() == null) {
+                activePatientProgramList.add(patientProgram);
+            }
+        }
+        return activePatientProgramList;
+    }
+
+    private boolean isNotInAnyConfiguredProgramState(ProgramWorkflowService programWorkflowService, PatientProgram activePatientProgram) {
+        return visitCloseData.getProgramStateConcepts().stream().noneMatch(concept -> {
+            List<ProgramWorkflowState> programWorkflowStatesByConcept = programWorkflowService.getProgramWorkflowStatesByConcept(concept);
+            ProgramWorkflowState programWorkFlowStateForConfiguredConcept = programWorkflowStatesByConcept.get(0);
+            PatientState patientState = activePatientProgram.getCurrentState(null);
+            ProgramWorkflowState patientCurrentWorkFlowState = patientState.getState();
+            return patientCurrentWorkFlowState.equals(programWorkFlowStateForConfiguredConcept) && patientState.getEndDate() == null;
+        });
+    }
+
+    private boolean isBedAssigned(Patient patient){
+        BedManagementService bedManagementService = Context.getService(BedManagementService.class);
+        BedDetails bedAssignmentDetailsByPatient = bedManagementService.getBedAssignmentDetailsByPatient(patient);
+        return !(bedAssignmentDetailsByPatient == null);
```

Comment:
@binduak @vinishad Any specific reason to use like this instead of directly using !=

Reply:
No specific reason, modified.